src/general/world_effects/UnderwaterVentEruptionEffect.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using Newtonsoft.Json;
+using Xoshiro.PRNG64;
+
+/// <summary>
+///   Underwater vents only event producing hydrogen sulfide and carbon dioxide
+/// </summary>
+[JSONDynamicTypeAllowed]
+public class UnderwaterVentEruptionEffect : IWorldEffect
+{
+    [JsonProperty]
+    private GameWorld targetWorld;
+
+    private XoShiRo256starstar random;
+
+    public UnderwaterVentEruptionEffect(GameWorld targetWorld, XoShiRo256starstar random)
+    {
+        this.targetWorld = targetWorld;
+        this.random = random;
```

Comment:
The random should be made into a read-only property with the JsonProperty attribute as it needs to be loaded from saves. Also as random instances aren't super safe to share, usually when a method takes in a random in Thrive, then in the method body I do `random = new Xosiro...(random.Next...)` to just take in a seed for the random from the other random but then not keep it around for safety reasons.